### PR TITLE
fix:  improve validation error message for Table widget

### LINKED ISSUE #28729: [Enhancement]: Improve error copy for validation errors
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

The current validation error message is hard to understand, and not actionable.

### Why should this be worked on?

This will improve the error resolution rate for validation errors

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description
This PR changes the validation error message for Table widgets

#### PR fixes following issue(s)
Fixes #28729
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
#### Type of change
- Bug fix (non-breaking change which fixes an issue)
>
>
#### How Has This Been Tested?
- [ ] Manual
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Refactor**
	- Improved the error message for table data validation to provide more specific guidance to users about the expected input format.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Binding/Invalid_binding_spec.js [hunk 1]
```diff
@@ -28,7 +28,9 @@ describe(
       //Table widget test with invalid binding value
       EditorNavigation.SelectEntityByName("Table1", EntityType.Widget);
       cy.testJsontext("tabledata", JSON.stringify(testdata.defaultdataBinding));
-      cy.evaluateErrorMessage(testdata.tableWidgetErrorMsg);
+      cy.evaluateErrorMessage(
+        `This value does not evaluate to type Array<Object>`,
+      );
```
[the remark this answers is not in the document]
Comment (ohansFavour) [reply app/client/cypress/e2e/Regression/ClientSide/Binding/Invalid_binding_spec.js:33]: Thus test is for SELECT Widgets

**Review by arunvjn — APPROVED** [review]

**Pull request merged**